RoslynPluginGenerator/NuGet/NuGetRepositoryFactory.cs

```diff
@@ -0,0 +1,68 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="NuGetRepositoryFactory.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using NuGet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SonarQube.Plugins.Roslyn
+{
+    /// <summary>
+    /// Creates the remote repository from which the packages will be fetched
+    /// </summary>
+    public static class NuGetRepositoryFactory
+    {
+        /// <summary>
+        /// Returns the settings loaded from the user- and machine-wide config files
+        /// </summary>
+        public static ISettings GetSettingsFromConfigFiles()
+        {
+            NuGetMachineWideSettings machineSettings = new NuGetMachineWideSettings();
+            ISettings settings = Settings.LoadDefaultSettings(null, null, machineSettings);
```

Comment:
The nulls here would allow us to specify another config file to look for if we wanted e.g. a config file in the same directory as executable. We can add this later if we think it would help.